packages/react-native-reanimated/src/component/ReduceMotionConfig.tsx

```diff
@@ -0,0 +1,33 @@
+'use strict';
+import { useEffect } from 'react';
+import { ReduceMotion } from '../commonTypes';
+import { IS_REDUCED_MOTION } from '../hook/useReducedMotion';
+
+/**
+ * A component that lets you overwrite default reduce motion behavior globally in your application.
+ *
+ * @param mode - Determines default reduce motion behavior globally in your application. Configured with {@link ReduceMotion} enum.
+ * @see https://docs.swmansion.com/react-native-reanimated/docs/components/ReduceMotionConfig
+ */
+export function ReduceMotionConfig({ mode }: { mode: ReduceMotion }) {
+  useEffect(() => {
+    if (!__DEV__) {
+      return;
+    }
+    console.warn(
+      `[Reanimated] Reduced motion setting is overwritten with mode '${mode}'.`
+    );
+  }, []);
+
+  useEffect(() => {
+    const wasEnabled = IS_REDUCED_MOTION.value;
```

Comment:
Looks like here we're reading a ~~shared value~~ mutable on the JS thread. and there are some performance concerns around that. Is it really safe to do so here?

Reply:
I believe it shouldn't have a significant impact, but I've thought of a new idea. Let me know what you think.